### ISSUE TITLE
`getStaticValue` ignores `Set`/`Map` mutation and folds `.size` to the initializer value

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [eslint-community/eslint-utils#353]
Tested versions: `4.10.1` & `5.0.0-alpha.5` (bug is present in both)

`getStaticValue` evaluates `Set.prototype.size` and `Map.prototype.size` (allowed via `getterAllowed`), but its mutation guard `hasMutationInProperty` only recognizes mutating **Array** methods. `Set.prototype.add`/`delete`/`clear` and `Map.prototype.set`/`delete`/`clear` are not detected, so the guard never fires and `.size` folds to the value from the initializer.

Found as causing https://github.com/sindresorhus/eslint-plugin-unicorn/issues/3607

**Reproduction**

```js
import {Linter} from 'eslint';
import {getStaticValue} from '@eslint-community/eslint-utils';

const code = `
const set = new Set();
set.add('a');
set.size;
`;

new Linter().verify(code, {
	plugins: {probe: {rules: {probe: {create: context => ({
		MemberExpression(node) {
			console.log(getStaticValue(node, context.sourceCode.getScope(node)));
		},
	})}}}},
	rules: {'probe/probe': 'error'},
});

```

**Expected:** `null` for `set.size` — the value is not statically determinable, since `set.add('a')` runs first.

**Actual:** `{value: 0}`.

The equivalent array case behaves correctly: `const list = []; list.push('a'); list.length` yields `null`, because `push` is in the mutator list. `Map` has the same defect as `Set` (`map.set(...)` then `map.size`).

**Cause**

-   `getterAllowed`  opts  `Map`/`Set`  `size`  into evaluation.
-   `hasMutationInProperty`  →  `isNameOfMutationArrayMethod`  checks only  `copyWithin`,  `fill`,  `pop`,  `push`,  `reverse`,  `shift`,  `sort`,  `splice`,  `unshift`.

So the property that mutation changes is readable, while the calls that change it are invisible to the guard.

**Comment by electrohyun** [comment]
Hi @ehoogeveen-medweb, I’ve reviewed the reproduction and root cause analysis provided in this issue. Are you planning to work on a fix? If not, would you be okay with me taking it on?

**Comment by ehoogeveen-medweb** [comment]
Go ahead! This issue isn't blocking me, so I'm not planning to work on a fix at this time.
